packages/devtools_app/lib/src/charts/flutter/flame_chart.dart

```diff
@@ -14,6 +18,162 @@ const double sectionSpacing = 15.0;
 const double topOffset = rowHeightWithPadding;
 const double sideInset = 70.0;
 
+abstract class FlameChart<T, V extends FlameChartNodeDataMixin>
```

Comment:
Rather than introducing the nodedatamixin, I suggest adding a `V selected` field, and a `onSelected(V)` callback for when an element in the chart is selected.

Whatever builds the flame chart can then manage the selections more easily.

Reply:
So long as we don't need to recompute the viewport regularly, rebuilding won't really be an issue.  The expensive part is repainting.

For storing selection state inside of widgets, what happens when a widget that is selected gets scrolled off the screen and disposed?  What happens to its selected state as it returns to the screen?

I'm fine with leaving a TODO, but we want to make state management on this chart simple and rebuilds fast.